### ISSUE TITLE
react build issue using nextjs framework 

### ISSUE THREAD

**Issue by Abdulla97** [Tap-Payments/goSell-checkout-sandbox-doc#1]
hello, 

I am facing build issue when npm package for GoSell 

![tap-next-err](https://user-images.githubusercontent.com/29528930/84052127-cf95d080-a9b8-11ea-99e3-b6003cf3a8b2.png)
  
I am not able to build because of Global css from node_modules  
 next team response 
https://github.com/vercel/next.js/blob/master/errors/css-npm.md

**Comment by anaszaman** [comment]
Its because of NextJS, find the file and add CSS into your _app.js

**Comment by Abdulla97** [comment]
yeah I already solve it by dynamic importing with disabling SSR in that component

**Comment by ahmed-abdelhamid** [comment]
@Abdulla97 
Do you really manage to solve that issue?
because even with dynamic import and disabling SSR it keeps crashes on the build process

**Comment by anaszaman** [comment]
Modified a little bit to work with NextJS, give it a try https://github.com/FassetIO/tap-payments

**Comment by ahmed-abdelhamid** [comment]
Ok great, but how I use it. Can I install it like a npm library or I have to make these modifications manually?

**Comment by anaszaman** [comment]
package.json
`"@tap-payments/gosell": "FassetIO/tap-payments",`

**Comment by ahmed-abdelhamid** [comment]
Thank you for your help

**Comment by Abdulla97** [comment]
> @Abdulla97
> Do you really manage to solve that issue?
> because even with dynamic import and disabling SSR it keeps crashes on the build process

hello 

add this to the payment componet index.js  
import dynamic from 'next/dynamic'

const DynamicComponentWithNoSSR = dynamic(
  () => import('./Payment'),
  { ssr: false }
)

function PaymentIndex() {
  return (
    <DynamicComponentWithNoSSR/>
  )
}

export default PaymentIndex

**Comment by BenHakimIlyass** [comment]
Hello dear @anaszaman, I've tried your solution and got another error which is:
![image](https://user-images.githubusercontent.com/50136109/131994178-a0b14cc5-b1e4-440e-bcc6-7ff9bb58f8d4.png)
 Any idea?

**Comment by FaysalMaqbool** [comment]
@Abdulla97  i have implemented the above code but it still giving the same issue

**Comment by FaysalMaqbool** [comment]
> package.json
> `"@tap-payments/gosell": "FassetIO/tap-payments",`

after adding this dependency it give the error
"window is not defined"

**Comment by anaszaman** [comment]
> > package.json
> > `"@tap-payments/gosell": "FassetIO/tap-payments",`
> 
> after adding this dependency it give the error
> "window is not defined"

Make sure to remove previous tap-payment module.

**Comment by FaysalMaqbool** [comment]
@anaszaman thanks it works.
But now GoSell.openLightBox(); not opening the payment page. it just not calling any loader or nor showing any activity. Can you please give me any pointer.

**Comment by FaysalMaqbool** [comment]
@anaszaman the error still exist after removing last module
first i run yarn remove @tap-payments/gosell
then i added "@tap-payments/gosell": "FassetIO/tap-payments", in my package.json and run yarn
it still gives me the same error even after removing cache and node-modules.

**Comment by BrodieCruz** [comment]
Has anyone fixed this problem? 
If I change version to "FassetIO/tap-payments" - i get "window is not defined" error.
How to make it work with NEXT??

**Comment by Doaa-Ismael** [comment]
I solved it  by using [this](https://www.npmjs.com/package/next-remove-imports) package, and wrapping GoSell component with another component so that I can lazy load it by using Next js dynamic imports.

**Comment by jk-baseer** [comment]
Hi Guys,
We are also facing the same issue in our nextjs project. @Doaa-Ismael we tried this but it didn't help. Can you tell us more on how to achieve this?

@AhmedElsharkawyCS is it possible your team can prepare a package for Nextjs or a sample repo for people to duplicate?

**Comment by anaszaman** [comment]
Looks like FassetIO org removed this package, I will check my local repos and try to publish under my name.

**Comment by jk-baseer** [comment]
@anaszaman Thanks bro, so what exactly is the issue? May be we can try from our end as well.

**Comment by anaszaman** [comment]
See if this works https://github.com/anaszaman/tap-payments, otherwise I will publish npm package.

**Comment by Ahsan-Ullah1871** [comment]
> See if this works https://github.com/anaszaman/tap-payments, otherwise I will publish npm package.

Hello,

I tried using it with your project, but I noticed that it's self-installing and not functioning properly. After removing all the node modules and attempting to reinstall, I found that it installed, but it's still not working. It's displaying an issue where the GoSell functions are not operational.

Upon inspecting your project's package.json file, I observed that an older version of the GoSell package is in use.
Is it feasible to update it to the latest version and consider publishing it as a package? 

That will be greatly helpful. Thank you in advance!
